examples/mistral/mistral-dpo.yml

```
@@ -0,0 +1,103 @@
+#Note that we are switching from the regular chat template to chatml.
+#If you experience problems with the special tokens, training for more epochs can help.
+#After trainig, merge the model before inference otherwise you might
+#face problems with the special tokens.
+
+base_model: mistralai/Mistral-7B-Instruct-v0.2
+model_type: MistralForCausalLM
+tokenizer_type: LlamaTokenizer
+
+load_in_8bit: false
+load_in_4bit: true
+strict: false
+
+chat_template: chatml
+rl: dpo
+datasets:
+  - path: olivermolenschot/alpaca_messages_dpo_test
+    type: chat_template.default
+    field_messages: conversation
+    field_chosen: chosen
+    field_rejected: rejected
+    message_field_role: role
+    message_field_content: content
+    roles:
+      system:
+        - system
+      user:
+        - user
+      assistant:
+        - assistant
+
+dataset_prepared_path:
+val_set_size: 0.05
+output_dir: ./outputs/dpo-lora-out-5
+
+sequence_len: 64
```

This is quite a low default? I suggest a bit higher, but feel free to adjust.
```suggestion
sequence_len: 2048
```

---

Yeah it is very low. I did this mainly to speed up training as my GPU only has 12GB of memory.

---

Would this not drop data points or are the sequences in the referred dataset really short?

---

The datapoints are quite short by design, so it is not dropping many rows. However, I tested it with max sequence length of 2048 and it also trained well. So I updated it and I think you are right, this would be a better example for others. 

I also made all the other changes you suggested and trained the model successfully.